app/uk/gov/hmrc/soletraderidentificationfrontend/utils/MessagesHelper.scala

```diff
@@ -18,12 +18,44 @@ package uk.gov.hmrc.soletraderidentificationfrontend.utils
 
 import play.api.http.HttpConfiguration
 import play.api.i18n.{DefaultMessagesApi, DefaultMessagesApiProvider, Langs}
-import play.api.libs.json.Json
 import play.api.{Configuration, Environment}
-import uk.gov.hmrc.soletraderidentificationfrontend.models.{JourneyConfig, TranslationLabels}
+import uk.gov.hmrc.soletraderidentificationfrontend.models.{JourneyConfig, JourneyLabels}
 
 import javax.inject.{Inject, Singleton}
 
+object MessagesHelper {
+
+  private val optServiceNameTranslationKey = "optServiceName"
+  private val optFullNameTranslationKey = "optFullNamePageLabel"
+
+  def amendMessagesWithLabelsFromJourneyConfig(initialMessages: Map[String, Map[String, String]],
+                                               journeyConfig: JourneyConfig): Map[String, Map[String, String]] = {
+
+    val extraEnglishTranslations: Map[String, String] = journeyConfig
+      .pageConfig
+      .labels match {
+      case Some(JourneyLabels(_, Some(optEnglishServiceName), _, Some(optEnlgishFullName))) => Map(optServiceNameTranslationKey -> optEnglishServiceName, optFullNameTranslationKey -> optEnlgishFullName)
+      case Some(JourneyLabels(_, None, _, None)) | None =>
+        journeyConfig.pageConfig.optServiceName.map(serviceName => Map(optServiceNameTranslationKey -> serviceName)).getOrElse(Map.empty)
+        journeyConfig.pageConfig.optFullNamePageLabel.map(fullName => Map(optFullNameTranslationKey -> fullName)).getOrElse(Map.empty)
```

Comment:
lines 39-40 do not match expected type: Map[String, String]
now they are two separate Map[String, String] and Map[String, String]